src/legendhpges/base.py

```diff
@@ -105,6 +111,107 @@ def _decode_polycone_coord(self) -> tuple[list[float], list[float]]:
         Must be overloaded by derived classes.
         """
 
+    def get_profile(self) -> tuple[list[float], list[float]]:
+        """Get the profile of the HPGe detector.
+
+        Returns
+        -------
+        (r.z)
+            two lists of r and z coordinates, respectively.
+
+        Note
+        -----
+            For V02160A and P00664A the detector profile is that of the solid without cut.
+        """
+        if isinstance(self.solid, geant4.solid.GenericPolycone) is False:
+            r = self.solid.obj1.pR
+            z = self.solid.obj1.pZ
+        else:
+            r = self.solid.pR
+            z = self.solid.pZ
+        return r, z
+
+    def is_inside(self, coords: ArrayLike, tol: float = 1e-11) -> NDArray[bool]:
+        """Compute whether each point is inside the shape
+
+        Parameters
+        ----------
+        coords
+            2D array of shape `(n,3)` of `(x,y,z)` coordinates for each of `n` points, second index corresponds to `(x,y,z)`.
+        tol
+            distance outside the surface which is considered inside.
+
+        Returns
+        -------
+        numpy array of booleans.
+        """
+
+        if isinstance(self.solid, geant4.solid.GenericPolycone) is False:
+            msg = f"distance_to_surface is not implemented for {type(self.solid)} yet"
+            raise NotImplementedError(msg)
+
+        if isinstance(self.solid, geant4.solid.GenericPolycone) is False:
```

Comment:
```suggestion
        if not isinstance(self.solid, geant4.solid.GenericPolycone):
```

check was duplicated?